### PR TITLE
Clean up names and export types

### PR DESCRIPTION
**Pull request by pajter**
- Fixed missing exports in CSS and JS. Some files were missing in the CSS and JS exports.
- Fixed `ramp` type in data-colors for JS.
- Fixed naming discrepancy between fonts and font-vars. LESS and SCSS were using `fonts`, while `fonts` was also being used for the actual @font-face rules. I ensured all exports are now using `font-vars` for the actual variables. This will break existing imports that use `fonts.[ext]`. They'll need to be renamed to `font-vars.[ext]`.

### CODE REVIEW

**Comment by kimhart** [comment]
Good call @pajter -- tagging @monibons on this for visibility.

**Comment by monibons** [comment]
Thanks @pajter, I have a couple of questions / concerns:

1. We intentionally removed the non-data color variable outputs to JS since data colors are the only colors that serve a purpose in JS (ie. data visualizations) and we'd like to keep this library slim and free of extra fat where possible. John did a lot of overthinking for scenarios we just haven't had a need for.

2. We also intentionally modified the data exports for `SCSS` / `Less` to be single variables versus lists because they were easier to use in practice. We did, however, leave the exports to `js` as ramps because they are easier to apply in javascript chart libraries as lists. 

If you're coming across scenarios or use cases where you need access to our brand/system/code colors in `js` then it might be valuable to discuss so we're all in the loop of those specific use cases.

**Comment by pajter** [comment]
> 1. We intentionally removed the non-data color variable outputs to JS since data colors are the only colors that serve a purpose in JS (ie. data visualizations) and we'd like to keep this library slim and free of extra fat where possible. John did a lot of overthinking for scenarios we just haven't had a need for.

Fair enough. I don't really see this is adding extra fat since they're ESM exports which are fully tree-shakeable. If the concern is file-size then that's negligible.

> 2. We also intentionally modified the data exports for `SCSS` / `Less` to be single variables versus lists because they were easier to use in practice. We did, however, leave the exports to `js` as ramps because they are easier to apply in javascript chart libraries as lists. 

That makes sense! Wasn't aware of that. I can put that back. :)

> If you're coming across scenarios or use cases where you need access to our brand/system/code colors in `js` then it might be valuable to discuss so we're all in the loop of those specific use cases.

Exporting to JS also provides the ability to use CSS-in-JS solutions. Just thinking that if it's not a lot of maintenance overhead (everything's fully automated), why not support as much as possible?

**Comment by monibons** [comment]
I see your point about "if we can then why not?" but I tend to lean towards taking a stand on what we would recommend as best or better practice. I know there are both advantages and disadvantages in the SCSS vs styled components debate but we seem to have taken a preference for staying the path of SCSS? I also do think there is some maintenance overhead that can slowly creep up as Kim and I recently found ourselves asking why all of those extra exports and files were necessary. It felt like overkill to generate assets that have never been used.

To be fully transparent, we've found patterns of bloat and over-engineering in John's projects so I may be a little biased or fatigued from that as we did other clean up in this library as well. I just have a strong preference for staying as lean as possible with support for realistic use cases versus hypothetical ones.

**Comment by pajter** [comment]
Yeah I'm definitely all for keeping things lean and clean. I personally don't see a few configs as huge overhead, but it's your call in the end as you're the main maintainers. :) I can revert some things. The only thing I needed for my use case was the missing CSS variables.

I do want to note that having no use cases can be a chicken-egg problem. If the intention is for the entire company to use it, then having more possibilities can lead to the use cases. Otherwise people will be more likely to just use Material Design or whatever else is available that works for them. I always try to look at it from an OSS perspective and try to be as inclusive as possible. Just my thoughts. :)

**Comment by monibons** [comment]
Valid point, + 10 for inclusivity. Consider me convinced ;) If you could just revert the `js` data color exports back to ramps aka lists then we can approve and merge.

Thanks for humoring the discussion @pajter !

**Comment by pajter** [comment]
Of course! Thank you for the friendly discussion. 😄

**Comment by pajter** [comment]
@monibons @kimhart I reverted the JS data colors type. :)

**Review by kimhart — APPROVED** [review]

**Pull request merged**
